### ISSUE TITLE
twisted.web.template.Tag.fillSlots documentation doesn't mention it returns self

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5288]
|[<img alt="bitglue's avatar" src="https://avatars.githubusercontent.com/u/1232472?s=50" width="50" height="50">](https://github.com/bitglue)| @bitglue reported|
|-|-|
|Trac ID|trac#5288|
|Type|enhancement|
|Created|2011-09-30 14:12:05Z|

.fillSlots() returns self. There is a test verifying this behavior, twisted.web.test.test_template.FlattenIntegrationTests.test_elementUsingSlots. The documentation should document this behavior, since this is a common idiom I have seen other people use:

```
def renderStuff(self, request, tag):
    return tag.fillSlots(something=self.something)
```

It probably also wouldn't hurt to add another test to make this more explicit.

Attachments:

* [template-fillSlots-documentation-5288.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/a7a/a7ae8d1e41c691a72f9f746c3dc482182001a7f7/b70e04b539e13eb03b17bd259aa62172ea9ce80b.patch) (1333 bytes) - added by indigo on 2011-09-30 15:24:17Z - proposed fix

<details><summary>Searchable metadata</summary>

```
trac-id__5288 5288
type__enhancement enhancement
reporter__indigo indigo
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__web web
keywords__documentation_easy documentation easy
time__1317391925000000 1317391925000000
changetime__1317406678000000 1317406678000000
version__None None
owner__exarkun exarkun

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun set owner to @exarkun|
|-|-|

Thanks.  Though it doesn't make a practical difference, I think `assertIdentical` expresses the intent of the test more precisely than `assertEqual` (no practical difference since `Tag` instances only compare equal to themselves).  Otherwise, the patch looks great.  I'll switch it over to `assertIdentical` and apply it.  Thanks again.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun set status to `closed`|
|-|-|

(In [32712]) Apply patch expanding `Tag.fillSlots` API documentation and adding an explicit test for its behavior.

Author: indigo
Reviewer: exarkun
Fixes: [#5288](https://github.com/twisted/twisted/issues/5288)

Explicitly document and unit test the return value (`self`) of
`twisted.web.template.Tag.fillSlots`.
